pkg/build/builder/docker.go

```diff
@@ -413,6 +419,33 @@ func traverseAST(cmd string, node *parser.Node) int {
 	return index
 }
 
+// insertEnvAfterFrom inserts an ENV Dockerfile instruction right after the
+// FROM instruction of the given dockerfile to define the environment variables
+// from env and returns the new Dockerfile.
+func insertEnvAfterFrom(env []kapi.EnvVar, dockerfile []byte) []byte {
+	if len(env) == 0 {
+		// Just return a copy.
+		return append([]byte(nil), dockerfile...)
+	}
+	fromLine := dockerfileFromLineRegexp.Find(dockerfile)
```

Comment:
It should, but as I mentioned earlier it is error-prone. Imagine a Dockerfile:

```
# FROM ubuntu
FROM centos
RUN ...
```

With the current implementation the output file would become:

```
# FROM ubuntu
ENV FOO=bar
FROM centos
RUN ...
```

We should (I will) make it more robust.